apps/api_web/lib/api_web/controllers/portal/user_controller.ex

```diff
@@ -20,18 +20,47 @@ defmodule ApiWeb.ClientPortal.UserController do
     |> render("new.html", changeset: changeset)
   end
 
+  def create(conn, %{"user" => user_params, "g-recaptcha-response" => recaptcha}) do
+    _create(conn, user_params, recaptcha)
+  end
+
+  # Allow missing recaptcha response
   def create(conn, %{"user" => user_params}) do
-    case ApiAccounts.register_user(user_params) do
-      {:ok, user} ->
-        conn
-        |> put_session(:user_id, user.id)
-        |> configure_session(renew: true)
-        |> redirect(to: portal_path(conn, :index))
+    _create(conn, user_params, nil)
```

Comment:
suggestion(non-blocking): You could combine these with `Map.get` which defaults to `nil`:

```elixir
def create(conn, %{"user" => user_params} = params) do
  recaptcha = Map.get(params, "g-recaptcha-response")
  ...
end
```